src/components/matrixSize.jsx

```diff
@@ -17,10 +17,33 @@ export default class MatrixSize extends React.Component {
       this.setState({ numCols: event.target.value });
   }
   handleClick(){
-  firebase.database().ref('grids/' + this.state.gridId + '/numRows' ).set(parseInt(this.state.numRows));
-  firebase.database().ref('grids/' + this.state.gridId + '/numCols' ).set(parseInt(this.state.numCols));
+  firebase.database().ref('grids/' + this.state.gridId).once('value', gridSnapshot => {
```

Comment:
I guess since you already put in the effort to do it this way and there is more benefit for this way, we should go ahead with what you have.